### PR TITLE
Add coverage age on simulant initialization

### PR DESCRIPTION
**Pull request by KjellSwedin**

### CODE REVIEW

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 1]
```diff
@@ -44,22 +43,27 @@ def setup(self, builder: 'Builder'):
...
                                                                   'iron_fortification.effective_coverage_level'],
                                                  requires_streams=[self._common_key])
+        builder.event.register_listener('time_step', self.on_time_step)
+
 
     def on_initialize_simulants(self, pop_data: 'SimulantData'):
         if pop_data.user_data['sim_state'] == 'setup':  # Initial population
             update_folic_acid = pd.Series('unknown', index=pop_data.index, name=self._fa_column)
             update_iron = pd.Series('unknown', index=pop_data.index, name=self._iron_column)
+            update_coverage_age_start = pd.Series(pd.NaT, index=pop_data.index, name=self._iron_coverage_age)
```
Comment (collijk) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py]: This should be a nan, it's not a date time

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 2]
```diff
@@ -70,9 +74,22 @@ def on_initialize_simulants(self, pop_data: 'SimulantData'):
             update_iron = pd.Series((draw < effective_coverage_iron).map({True: 'covered', False: 'uncovered'}),
                                    index=pop_data.index,
                                    name=self._iron_column)
-        pop_update = pd.concat([update_folic_acid, update_iron], axis=1, keys=[self._fa_column, self._iron_column])
+            update_coverage_age_start = pd.Series((update_folic_acid=='covered').map({True: 0.0, False: pd.NaT}),
```
Comment (collijk) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py]: Same here

Comment (collijk) [reply src/vivarium_conic_lsff/components/fortification/folic_acid.py]: The age start should be mapped from iron, not folic acid as well

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 3]
```diff
@@ -70,9 +74,22 @@ def on_initialize_simulants(self, pop_data: 'SimulantData'):
...
+
+        pop_update = pd.DataFrame({
+            self._fa_column: update_folic_acid,
+            self._iron_column: update_iron,
+            self._iron_coverage_age: update_coverage_age_start
+        }, pop_data.index)
         self.population_view.update(pop_update)
 
+
+    def on_time_step(self, event: 'Event'):
```
Comment (collijk) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py]: Put a pass here for now.

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 4]
```diff
@@ -44,35 +45,76 @@ def setup(self, builder: 'Builder'):
...
 
     def on_initialize_simulants(self, pop_data: 'SimulantData'):
         if pop_data.user_data['sim_state'] == 'setup':  # Initial population
             update_folic_acid = pd.Series('unknown', index=pop_data.index, name=self._fa_column)
-            update_iron = pd.Series('unknown', index=pop_data.index, name=self._iron_column)
+            update_iron = pd.Series('unknown', index=pop_data.index, name=self._iron_fortified_column)
+
+            propensity = self.randomness.get_draw(pop_data.index)
+            is_covered = self.is_covered(propensity)
+            update_iron_coverage_age_start = pd.Series((is_covered).map({True: 0.0, False: np.nan}),
```
Comment (KjellSwedin) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py]: Clarify (mother_folic, mother_iron, individual)

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 5]
```diff
@@ -44,35 +45,76 @@ def setup(self, builder: 'Builder'):
...
                                                                   'iron_fortification.effective_coverage_level'],
                                                  requires_streams=[self._common_key])
+        builder.event.register_listener('time_step', self.on_time_step)
+
 
     def on_initialize_simulants(self, pop_data: 'SimulantData'):
         if pop_data.user_data['sim_state'] == 'setup':  # Initial population
             update_folic_acid = pd.Series('unknown', index=pop_data.index, name=self._fa_column)
-            update_iron = pd.Series('unknown', index=pop_data.index, name=self._iron_column)
+            update_iron = pd.Series('unknown', index=pop_data.index, name=self._iron_fortified_column)
```
Comment (KjellSwedin) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py]: Clarify names (maternal versus individual)

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 6]
```diff
@@ -44,35 +45,76 @@ def setup(self, builder: 'Builder'):
...
+            propensity = self.randomness.get_draw(pop_data.index)
+            is_covered = self.is_covered(propensity)
+            update_iron_coverage_age_start = pd.Series((is_covered).map({True: 0.0, False: np.nan}),
+                                                       index=pop_data.index,
+                                                       name=self._iron_coverage_age)
+            pop_update = pd.DataFrame({
+                self._iron_fort_propensity: propensity,
+            }, pop_data.index)
+            self.population_view.update(pop_update)
         else:  # New sims
```
Comment (KjellSwedin) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py:78]: Move call to update to the end of the method

File: src/vivarium_conic_lsff/components/fortification/folic_acid.py [hunk 7]
```diff
@@ -44,35 +45,76 @@ def setup(self, builder: 'Builder'):
...
 
+
+    def on_time_step(self, event: 'Event'):
+        """Update coverage start age for all newly covered individuals."""
+        pop = self.population_view.get(event.index, query='tracked == True and alive=="alive"')
+        is_covered = self.is_covered(pop[self._iron_fort_propensity])
+        not_previously_covered = pop[self._iron_coverage_age].isna()
+        newly_covered = is_covered & not_previously_covered
+        pop.loc[newly_covered, self._iron_coverage_age] = 0.0
+        self.population_view.update(pop)
```
Comment (KjellSwedin) [inline src/vivarium_conic_lsff/components/fortification/folic_acid.py:108]: This should be actual age, pop.age

**Pull request merged**